spring-cloud-kubernetes-fabric8-autoconfig/src/main/java/org/springframework/cloud/kubernetes/fabric8/Fabric8PodUtils.java

```diff
@@ -55,33 +55,25 @@ public Fabric8PodUtils(KubernetesClient client) {
 
 		this.client = client;
 		this.hostName = System.getenv(HOSTNAME);
-		this.current = LazilyInstantiate.using(() -> internalGetPod());
+		this.current = LazilyInstantiate.using(this::internalGetPod);
 	}
 
 	@Override
 	public Supplier<Pod> currentPod() {
 		return this.current;
 	}
 
-	@Override
-	public Boolean isInsideKubernetes() {
-		return currentPod().get() != null;
-	}
-
-	private synchronized Pod internalGetPod() {
```

Comment:
does not have to be `synchronized`, `LazilyInstantiate::get` already takes care of that